cmd/wasm-client/main.go

```diff
@@ -202,10 +216,28 @@ func (w *wasmClient) ConnectServer(_ js.Value, args []js.Value) interface{} {
 					),
 				)
 			}, func(data []byte) error {
+				parts := strings.Split(string(data), ": ")
+				if len(parts) != 2 || parts[0] != "Macaroon" {
+					return fmt.Errorf("authdata does " +
+						"not contain a macaroon")
+				}
+
+				macBytes, err := hex.DecodeString(parts[1])
+				if err != nil {
+					return err
+				}
+
+				mac := &macaroon.Macaroon{}
+				err = mac.UnmarshalBinary(macBytes)
+				if err != nil {
+					return fmt.Errorf("unable to decode "+
+						"macaroon: %v", err)
+				}
+
+				w.mac = mac
+
 				return callJsCallback(
-					w.cfg.OnAuthData, hex.EncodeToString(
-						data,
-					),
+					w.cfg.OnAuthData, string(data),
```

Comment:
Ah, so the hex encoding here was a mistake in the first place? I guess nobody looked at it more closely before...

Reply:
yeah. my mistake!